js/gemini.js

```diff
@@ -381,14 +392,37 @@ module.exports = class gemini extends Exchange {
         //     ]
         //
         const result = [];
+        let promises = [];
         for (let i = 0; i < response.length; i++) {
-            const marketId = response[i];
-            const market = marketId;
-            const idLength = marketId.length - 0;
-            const baseId = marketId.slice (0, idLength - 3);
-            const quoteId = marketId.slice (idLength - 3, idLength);
+            let marketId = response[i];
+            marketId = marketId.toUpperCase ();
+            const method = 'publicGetV1SymbolsDetailsSymbol';
+            const request = {
+                'symbol': marketId,
+            };
+            //
+            //     {
+            //         "symbol": "BTCUSD",
+            //         "base_currency": "BTC",
+            //         "quote_currency": "USD",
+            //         "tick_size": 1E-8,
+            //         "quote_increment": 0.01,
+            //         "min_order_size": "0.00001",
+            //         "status": "open",
+            //         "wrap_enabled": false
+            //     }
+            //
+            promises.push (this[method] (this.extend (request, params)));
+        }
+        promises = await Promise.all (promises);
+        for (let i = 0; i < promises.length; i++) {
```

Comment:
@pcriadoperez Unfortunately I don't think we can afford 142 requests in a single call, it is too much. We should either load only the symbols  (base + quote) using `publicGetV1Symbols` or hardcode the values in options and use it.

Reply:
@carlosmiei hehe I had a feeling you might say that. However we do need the details like the precision to do things like create an order. In my opinion I think it's better having a heavier call than introducing another option that the user has to be aware of, but let me know if you think it's the way to go and I can add it.

Reply:
@pcriadoperez the thing is, some users might be using paid proxies and stuff like that and we don't want to bust their limits just to load markets. Btw, how does this related to the `watchOrders` problem? We were scraping the markets, right? it's not pretty I know but I don't understand why it impacts the WS functionality

Reply:
@carlosmiei The watchOrders I was getting a Network error when using the scrapper to fetchMarkets, that's what made me look into this. I'll look more into why the scrapper wasn't working and allow the passing the symbols as an option when using the API

Reply:
@pcriadoperez, what is the reason behind the failure? If it fails 50% of the time, what if we added a retry mechanism?

Reply:
I'm not too sure, fetch throws an internal `fetch failed` due to `TypeError` by nodejs/undici. I tried to search a little about it and seems to disappear when I switch to node v16

Reply:
@pcriadoperez but just in case, can we add the retry mechanism? I mean, if it works after 2 tries still offsets the alternative by 140 requests

Reply:
@pcriadoperez can you check this comment plssss ⬆️

Reply:
Hi @carlosmiei , I looked more into this, because I wanted to see if there was a cleaner solution to the retry.
It looks like it only happens in the latest 19 version of node due to a bug in undici. And a fix will be released very soon (https://github.com/nodejs/undici/issues/1776)
In node  v16 and v18 LTS it works fine, also in php and python. So not sure it's worth adding the retry, thoughts?

Reply:
Hey @pcriadoperez, thanks for checking it out,

I still think we should add the re-try because people might stick with buggy versions for a long time, and I don't think it's hard to do just a while loop and a try catch.